File: src/main/java/eu/crowdliterature/CrowdService.java
package eu.crowdliterature;

import java.util.List;

import javax.ws.rs.PathParam;

import de.deepamehta.core.JSONEnabled;
import eu.crowdliterature.model.Event;
import eu.crowdliterature.model.Institution;
import eu.crowdliterature.model.InstitutionOfMap;
import eu.crowdliterature.model.Person;
import eu.crowdliterature.model.PersonOfMap;
import eu.crowdliterature.model.Work;



public interface CrowdService {

    String getStartPageContent();
    
    SearchResult searchTopics(String searchTerm);

    // --- Event ---
    
	Event getEvent(long eventId);

    
    // --- Person ---

    Person getPerson(long personId);

    List<PersonOfMap> getAllPersons();

    // Editable Person
    void ensureUserToPersonAssociationAndWorkspaceSetup();
    
    long getPersonIdByLoggedInUser();
    
    // --- Work ---

    Work getWork(long workId);

    // --- Institution ---

    Institution getInstitution(long instId);

    List<InstitutionOfMap> getAllInstitutions();
    
    interface SearchResult extends JSONEnabled {}
}

File: src/main/resources/web/services/crowd-service.js
angular.module("crowd").service("crowdService", function($http) {

    this.getStartPageContent = function(callback) {
        $http.get("/crowd/start_page").then(callback);
    }

    this.loadBustourGeojson = function(callback) {
        $http.get("crowd-bustour.geo.json").then(callback);
    }

    // --- Person ---

    this.getPerson = function(personId, callback) {
        $http.get("/crowd/person/" + personId).then(callback);
    }

    this.getAllPersons = function(callback) {
        return $http.get("/crowd/persons").then(callback);
    }

    // --- Work ---

    this.getWork = function(workId, callback) {
        $http.get("/crowd/work/" + workId).then(callback);
    }

    // --- Institution ---

    this.getInstitution = function(instId, callback) {
        $http.get("/crowd/institution/" + instId).then(callback);
    }

    this.getAllInstitutions = function(callback) {
        return $http.get("/crowd/institutions").then(callback);
    }

})
